interpreter/interpreter.go

```diff
@@ -794,10 +795,18 @@ func (f function) resolveIdentifiers(n semantic.Node) (semantic.Node, error) {
 			}
 		}
 		v, ok := f.scope.Lookup(n.Name)
-		if !ok {
-			return nil, fmt.Errorf("name %q does not exist in scope", n.Name)
+		if ok {
```

Comment:
As written in the review comment, I would test the external scope after `localIdentifiers`.

Also, why don't you use a `NestedScope` for local identifiers instead of a slice?